### PR TITLE
[CET-1072] Fix entity page undefined error

### PR DESCRIPTION
**Pull request by nikhilunni**
## Change description

* Also fix division by 0 error in InitiativeStats if there are no services

## Type of change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Improvement (improves codebase without changing functionality)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: src/components/EntityPage/EntityPage.tsx [hunk 1]
```diff
@@ -49,13 +50,13 @@ export const EntityPage = () => {
   );
 
   useEffect(() => {
-    const scorecardId = initialScorecardId ?? scores?.[0].scorecard.id;
+    const scorecardId = initialScorecardId ?? scores?.[0]?.scorecard?.id;
```
Comment (nikhilunni) [inline src/components/EntityPage/EntityPage.tsx:53]: this was the bug

File: src/components/EntityPage/EntityPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * limitations under the License.
+ */
+import { CortexApi } from "../../api/CortexApi";
+import React from "react";
+import { Fixtures, renderWrapped } from "../../utils/TestUtils";
+import { EntityPage } from "./EntityPage";
+import { EntityProvider } from "@backstage/plugin-catalog-react";
+
+describe('EntityPage', () => {
+
```
Comment (cristinabuenahora) [inline src/components/EntityPage/EntityPage.test.tsx:23]: nit: empty line not needed here

File: src/components/EntityPage/EntityPage.tsx [hunk 3]
```diff
@@ -65,7 +66,7 @@ export const EntityPage = () => {
 
   if (scoresError || scores === undefined) {
     return (
-      <WarningPanel severity="error" title="Could not load scorecards.">
+      <WarningPanel severity="error" title="Could not load Scorecards.">
```
Comment (cristinabuenahora) [inline src/components/EntityPage/EntityPage.tsx:69]: nice

**Review by cristinabuenahora — APPROVED** [review]

File: src/utils/NumberUtils.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * @param numerator Numerator
+ * @param denominator Denominator
+ */
+export const safeDivide = (numerator: number, denominator: number) => {
+  if (denominator === 0) {
+    return 0
+  } else {
+    return numerator / denominator
+  }
+}
```
Comment (FoodProduct) [inline src/utils/NumberUtils.ts:28]: nit: eof newlines

File: yarn.lock [hunk 5]
```diff
@@ -3331,6 +3331,13 @@
   resolved "https://registry.npmjs.org/@types/btoa-lite/-/btoa-lite-1.0.0.tgz#e190a5a548e0b348adb0df9ac7fa5f1151c7cca4"
```
Comment (FoodProduct) [inline yarn.lock:3331]: Does `npx yarn-deduplicate` do anything here? 🙂

**Review by FoodProduct — DISMISSED** [review]

**Pull request merged**